cmd/service/service.go

```diff
@@ -42,10 +41,10 @@ func startServer() {
 		api.NewUpgrader(actionUpgrade),
 		api.NewHistory(actionHistory))
 
-	router.Handle("/ping", setContentType(ping.Handler()))
-	router.Handle("/list", setContentType(api.List(service)))
-	router.Handle("/install", setContentType(api.Install(service)))
-	router.Handle("/upgrade", setContentType(api.Upgrade(service)))
+	router.Handle("/ping", ContentTypeMiddle(api.Ping())).Methods(http.MethodGet)
+	router.Handle("/list", ContentTypeMiddle(api.List(service))).Methods(http.MethodPost)
```

Comment:
list should be `GET` as we don't have any changes / side effect also it's idempotent in a way.